src/api.rs

```diff
@@ -52,6 +58,9 @@ impl From<&str> for Flags {
                 's' => {
                     result.dot_all = true;
                 }
+                'u' => {
```

Comment:
I removed the 'u' flag and adjusted my new tests accordingly.

Just FYI and for some context; I'm always testing on my local fork of boa to see how my changes affect the ecma262 test suite. With the 'u' flag I get +100 passing tests on this branch, without it I "only" get +54.

Of course you are right. If anything ,this branch is not the right place to introduce something like that. I think at some point in the future I will make an effort to maybe introduce the 'u' flag to make regress and boa more spec compliant. Maybe it could even be a reverse flag ('non-unicode') to introduce 'legacy' behavior. But that is for another time ;)